module-4/backend/week_four.md

```diff
@@ -1,20 +1,63 @@
+
 * Why do we create staging environments?
+
+Staging environments allow developers to have a place to test their features in a production-like environment before the environment is actually deployed to production and its users. 
+
 * What are the characteristics of a good README (in your opinion)?
+
+* Project configuration and installation instructions (clone this, bundle that, etc.) 
+* Contact information for the distributor or programmer
+* Common issues and troubleshooting steps
+* Known bugs
+* Instructions to submit bugs, feature requests, submit patches, join mailing list, get announcements, or join the user or dev community in other forms
+* Credits and Acknowledgements
+* Copyright and licensing information
+
 * What's one main improvement you're going to make to your code regarding accessibility issues?
+
+I will definitely make sure to be using links and buttons that are "tab-able", it's definitely something I didn't do in Quantified Self, and it can easily be implemented using the tabindex if I don't use an <a> or <button> tag.
+
 * What are some basic security concerns to be aware of when building applications?
+
+* SQL and JavaScript Injection - when a user injects SQL and JS into user inputs, so a best practice is never trust user input and interpolate directly - best to strip out SQL before sending to the DB and any <script> tags in post requests from users.
+* Click Jacking - employed on malicious websites to make you do things on other websites - common prevention measure is to prevent the browser from allowing your site to be loaded in an iframe.
+
 * Why is continuous integration helpful/important?
+
+Continuous Integration allows us to create an environment purely for testing. I've seen it be immensely helpful at my old company where Pull Requests automatically ran the CI test suite and were "unmergeable" unless the code passed all tests.
+
 * What are some continuous integration tools?
+  * Travis CI
+  * Circle
+  * Code Climate
 
 #### Review  
 
 * What are some characteristics of "good" git workflow?
+
+Especially at Turing, some characteristics of a good git workflow include:
+* Creating new branches for new features
+* Making PRs that are reviewed and merged by others (if on a group project!)
+* Pulling down from master and merging changes locally before making a new PR 
+* Running test suite and ensuring all tests pass after merging and before making a new PR
+
 * What are the four fundamental concepts of object oriented programming?
+
+1. Abstraction - objects are reperesneted as real-world nouns and can be given methods that describe the behavior the programmer is trying to represent
```

Comment:
This is also related to SRP. Creating objects and methods that are flexible and not bound to specific cases is key.